testsuites/syncgateway/upgrade/testserver/sg_replicate/test_upgrade_with_sgreplicate.py

```diff
@@ -0,0 +1,611 @@
+import os
+import random
+import time
+
+from concurrent.futures import ProcessPoolExecutor
+from couchbase.bucket import Bucket
+from keywords.couchbaseserver import verify_server_version
+from keywords.utils import log_info, host_for_url
+from keywords.SyncGateway import (verify_sync_gateway_version,
+                                  verify_sync_gateway_product_info,
+                                  SyncGateway)
+from keywords.ClusterKeywords import ClusterKeywords
+from keywords.MobileRestClient import MobileRestClient
+from keywords.constants import SDK_TIMEOUT
+from utilities.cluster_config_utils import persist_cluster_config_environment_prop, replace_string_on_sgw_config, copy_sgconf_to_temp
+
+from libraries.testkit.cluster import Cluster
+from keywords import attachment
+from CBLClient.Authenticator import Authenticator
+from CBLClient.Document import Document
+from CBLClient.Replication import Replication
+from keywords.SyncGateway import sync_gateway_config_path_for_mode, setup_sgreplicate1_on_sgconfig, setup_replications_on_sgconfig
+from utilities.cluster_config_utils import load_cluster_config_json
+
+
+def test_upgrade(params_from_base_test_setup, setup_customized_teardown_test):
+    """
+    @summary
+        This test case validates Sync Gateway server to be upgraded
+        wihtout replication downtime along with couchbase lite clients
+        [test prerequisites]:
+        The conftest prepares the test environment,
+        the initial (old) versions of SGW and CBS are provisioned at this point
+        [test environment configuration]:
+        The test environment setup includes: 2 CBS, 2SGW, and 1 SGW load balancer
+        The SGW load balancer is configured with nginx
+        [test scenarios and steps]:
+        1. Create user, session and docs on SG
```

Comment:
Add any document link that you have created for this test

Reply:
Thanks for pointing out. I realized that summary for sg-replicate2 not updated. I have updated the summary and pointed out testplan